src/libraries/System.Text.Json/Common/JsonSeparatorNamingPolicy.cs

```diff
@@ -0,0 +1,157 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Buffers;
+using System.Globalization;
+
+namespace System.Text.Json
+{
+    internal abstract class JsonSeparatorNamingPolicy : JsonNamingPolicy
+    {
+        private readonly bool _lowercase;
+        private readonly char _separator;
+
+        internal JsonSeparatorNamingPolicy(bool lowercase, char separator) =>
+            (_lowercase, _separator) = (lowercase, separator);
+
+        public sealed override string ConvertName(string name)
+        {
+            // Rented buffer 20% longer that the input.
+            int bufferLength = (12 * name.Length) / 10;
```

Comment:
For smaller lengths (i.e. < 5) the buffer will essentially be the same as the string length. Perhaps the formula should ensure that the target buffer is padded by at least 2-5 characters to ensure resizes don't take place in most cases?

Reply:
It already happens by using `stackalloc char[JsonConstants.StackallocCharThreshold]`.

Reply:
Got it. For some reason I assumed you were slicing the buffer up to `bufferLength` but clearly that's not necessary.

Reply:
But.. if that is the case then you would need to update the `ExpandBuffer` function because currently it is doubling based on the stated value of `bufferLength`. I suspect that in some cases you could end up with a rented buffer that is smaller than the original stack allocated buffer.